src/julia_internal.h

```diff
@@ -497,8 +497,20 @@ STATIC_INLINE jl_value_t *jl_gc_alloc_(jl_ptls_t ptls, size_t sz, void *ty)
     const size_t allocsz = sz + sizeof(jl_taggedvalue_t);
     if (sz <= GC_MAX_SZCLASS) {
         int pool_id = jl_gc_szclass(allocsz);
-        jl_gc_pool_t *p = &ptls->heap.norm_pools[pool_id];
         int osize = jl_gc_sizeclasses[pool_id];
+        jl_gc_pool_t *p = &ptls->heap.norm_pools[pool_id];
+        if (ty == jl_simplevector_type) {
+            p += JL_GC_N_MAX_POOLS;
```

Comment:
are all `simple_vector`s long lasting? I think a number of them likely are thrown out pretty quickly.

Reply:
> I think a number of them likely are thrown out pretty quickly.

Possibly. I'm still curating the list of which objects should be special cased in the pool allocator.